src/org/rust/lang/core/psi/impl/RustItemImpl.kt

```diff
@@ -0,0 +1,18 @@
+package org.rust.lang.core.psi.impl
+
+import com.intellij.lang.ASTNode
+import org.rust.lang.core.psi.RustCompositeElementTypes
+import org.rust.lang.core.psi.RustItem
+import org.rust.lang.core.psi.RustOuterAttr
+import org.rust.lang.core.psi.RustVis
+
+public abstract class RustItemImpl(node: ASTNode)   : RustNamedElementImpl(node)
+                                                    , RustItem {
+
+    override val attrs: List<RustOuterAttr>?
+        get() = findChildrenByType(RustCompositeElementTypes.OUTER_ATTR)
+
+    override val vis: RustVis?
```

Comment:
Keep in mind that `pub` can be applied not only to items, but also to a struct field:

```Rust
pub struct S {
    pub foo: i32,
    bar: i32
}
```

It might be convinient to extract  a separate `RustVisibleThing` interface.